tests/points/validator_tests.py

```diff
@@ -0,0 +1,180 @@
+import json
+import unittest
+from plenario import create_app
+
+
+class TestValidator(unittest.TestCase):
+
+    def get_json_response_data(self, endpoint):
+        """A little util that does work I found myself repeating alot."""
+
+        response = self.test_client.get('/v1/api/' + endpoint)
+        return json.loads(response.data)
+
+    def setUp(self):
+        self.app = create_app()
+        self.test_client = self.app.test_client()
+
+    def test_validator_bad_dataset_name(self):
+        endpoint = 'detail'
+        query = '?dataset_name=crimez&obs_date__ge=2000'
+
+        resp_data = self.get_json_response_data(endpoint + query)
+
+        self.assertTrue(len(resp_data['meta']['message']) == 1)
+        self.assertEquals(resp_data['meta']['message']['dataset_name'], ['Not a valid choice.'])
+
+    def test_validator_bad_dataset_name_and_date(self):
+        endpoint = 'detail'
+        query = '?dataset_name=crimez&obs_date__ge=20z00'
+
+        resp_data = self.get_json_response_data(endpoint + query)
+
+        self.assertTrue(len(resp_data['meta']['message']) == 2)
+        self.assertEquals(resp_data['meta']['message']['dataset_name'], ['Not a valid choice.'])
+        self.assertEquals(resp_data['meta']['message']['obs_date__ge'], ['Not a valid date.'])
+
+    def test_validator_bad_column_name(self):
+        endpoint = 'detail'
+        query = '?dataset_name=crimes&obs_date__ge=2000&fake_column=fake'
+
+        resp_data = self.get_json_response_data(endpoint + query)
+
+        self.assertTrue(len(resp_data['meta']['message']) == 2)
+        self.assertTrue('Unused param' in resp_data['meta']['message'][0])
+        self.assertTrue('not a valid column' in resp_data['meta']['message'][1])
+
+    def test_validator_incorrect_datatype(self):
+        endpoint = 'detail'
+        query = '?dataset_name=crimes&obs_date__ge=2000&data_type=fake'
+
+        resp_data = self.get_json_response_data(endpoint + query)
+
+        self.assertTrue(len(resp_data['meta']['message']) == 1)
+        self.assertEquals(resp_data['meta']['message']['data_type'], ['Not a valid choice.'])
+
+    def test_shape_validator_incorrect_datatype(self):
+        endpoint = 'shapes/pedestrian_streets'
+        query = '?data_type=csv'
+
+        resp_data = self.get_json_response_data(endpoint + query)
+
+        self.assertTrue(len(resp_data['meta']['message']) == 1)
+        self.assertEquals(resp_data['meta']['message']['data_type'], ['Not a valid choice.'])
+
+    def test_validator_with_good_column_but_bad_value(self):
+        endpoint = 'detail'
+        query = '?dataset_name=crimes&data_type=json&id="break_me"'
+
+        resp_data = self.get_json_response_data(endpoint + query)
+
+        self.assertTrue(len(resp_data['meta']['message']) == 2)
+        self.assertTrue('Unused param' in resp_data['meta']['message'][0])
+        self.assertTrue('not a valid value' in resp_data['meta']['message'][1])
```

Comment:
I don't think we should check for the exact text of error messages in our tests. Making a more descriptive error message would break the test. I prefer to check for HTTP status code.